utils_nlp/models/transformers/common.py

```diff
@@ -11,7 +11,14 @@
 import numpy as np
 import torch
 from tqdm import tqdm, trange
-from transformers import AdamW, WarmupLinearSchedule
+from transformers import AdamW
+try:
```

Comment:
Can we simply import transformers and use transformers.__version__ to check its version?

Reply:
If we use `transformers.__version__`, we can only get a string but there are many possibilities. The break change happened at version '2.2.0' but we cannot simply check the version string that all version lower than '2.2.0' should use `WarmupLinearSchedule`, otherwise use `get_linear_schedule_with_warmup`. 

I think catch the import error should be an easier solution. It will compatible with all version.

Reply:
But it's not guaranteed lower versions will have `linear_schedule_with_warmup`.
I'm going to set the version requirement for transformers to >=2.1.1 so that it's guaranteed one of  `linear_schedule_with_warmup` and `WarmupLinearSchedule` is available. 
For version checking, I guess either way is fine. Just thought the transformer.\_\_verion\_\_ method is simpler.

Reply:
You don't need to specify the version and it's also not recommended. The code will handle all versions. I used `TRANSFORMERS_VERSION` as a marker (even it's a string variable to show which version had this break change).

Reply:
I didn't pin the version to a specific number, just to make sure it's greater than 2.1.1, so that when `linear_schedule_with_warmup` is not available, `WarmupLinearSchedule` is available.